hazelcast/src/main/java/com/hazelcast/map/impl/EntryViews.java

```diff
@@ -52,17 +53,32 @@ public static <K, V> EntryView<K, V> createSimpleEntryView(K key, V value, Recor
 
     public static <K, V> WanMapEntryView<K, V> createWanEntryView(Data key, Data value,
                                                                   Record<V> record, ExpiryMetadata expiryMetadata,
-                                                                  SerializationService serializationService) {
+                                                                  SerializationService serializationService,
+                                                                  boolean isPerEntryStatsEnabled) {
         return new WanMapEntryView<K, V>(key, value, serializationService)
                 .withCost(record.getCost())
                 .withVersion(record.getVersion())
                 .withHits(record.getHits())
                 .withLastAccessTime(record.getLastAccessTime())
-                .withLastUpdateTime(record.getLastUpdateTime())
+                .withLastUpdateTime(calculateLastUpdateTime(isPerEntryStatsEnabled, record, expiryMetadata))
                 .withCreationTime(record.getCreationTime())
                 .withLastStoredTime(record.getLastStoredTime())
                 .withTtl(expiryMetadata.getTtl())
                 .withMaxIdle(expiryMetadata.getMaxIdle())
                 .withExpirationTime(expiryMetadata.getExpirationTime());
     }
+
+    private static <V> long calculateLastUpdateTime(boolean isPerEntryStatsEnabled,
+                                                    Record<V> record,
+                                                    ExpiryMetadata expiryMetadata) {
+        if (isPerEntryStatsEnabled) {
+            return record.getLastUpdateTime();
+        }
+
+        if (expiryMetadata != ExpiryMetadata.NULL) {
+            return expiryMetadata.getLastUpdateTime();
+        }
```

Comment:
```suggestion
        if (expiryMetadata != ExpiryMetadata.NULL) {
            return expiryMetadata.getLastUpdateTime();
        }
        if (isPerEntryStatsEnabled) {
            return record.getLastUpdateTime();
        }
```

Seems this order reflects intention better.

Reply:
https://github.com/hazelcast/hazelcast/pull/23279/commits/fc2720695ff701179f9c306ae395d23265ecfd68